src/gpuarray_array.c

```diff
@@ -574,11 +574,11 @@ int GpuArray_setarray(GpuArray *a, const GpuArray *v) {
     }
   }
 
-
   memcpy(&tv, v, sizeof(GpuArray));
   tv.nd = a->nd;
   tv.dimensions = a->dimensions;
   tv.strides = strs;
+  tv.flags &= ~(GA_C_CONTIGUOUS|GA_F_CONTIGUOUS);
```

Comment:
Since we change the dimensions the array is no longer contiguous and the code would do the wrong thing if we left those flags.